sites/topomath.asu.edu/modules/system_descriptions/system_descriptions.module

```diff
@@ -0,0 +1,118 @@
+<?php
+
+function system_descriptions_form_alter(&$form, &$form_state, $form_id){
+	switch($form_id){ 
+		case "problem_form": //adds the system description button with weight -5 so that it appears just after the
+			$form['sys-desc-inp'] = array(
+				'#type' => 'button',
+				'#value' => 'System descriptions',
+				'#id' => 'open_sysdescmodal',
+				'#display' => 'none',
+				'#weight' => '-5'
+			); 
+			$form['#attached']['js'] = array(
+				drupal_get_path('module', 'system_descriptions') . '/static/sys_desc.js',
+			);
+			break;
+		}
+	}
+
+/* creates additional blocks which are specific to topomath in context of system descriptions*/
+function createSpecificBlocks(){
+	$specific_blocks = '<div id="handleSystemDesc" class="modal fade" role="dialog">
+								<div class="modal-dialog">
+								<!-- Modal content-->
```

Comment:
Converted two major components ( 1. All the non class model operations related buttons 2. System descriptions pane which manages all sys desc operations ) into drupal blocks on models page. The other category of snippets are bootstrap modals which all can be created dynamically based on part of the LMS user is using. I think that should be a separate module and we should create another separate task for it